.github/workflows/release_notes.yml

```diff
@@ -0,0 +1,69 @@
+name: Generate Release Notes
+
+on:
+  pull_request:
+
+# on:
+#   push:
+#     branches:
+#       - main
+
+jobs:
+  send-pull-requests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.1
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Generate release notes and environment variables
+        run: |
+          ruby -r /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/.github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.write_output_file
+          echo "file_name=$(ruby -r /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/.github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.file_name)" >> $GITHUB_ENV
```

Comment:
What if the `write_output_file` method returned the filename that it wrote to instead of the result of the write operation? Could you combine the two separate "write file" and "get filename" operations into one if you did that?

Reply:
I think we could! Do you think that would muddy the human readability of what is happening when it comes to assigning the Github variables?

Reply:
I think the storing of the result into an env var named "file_name" still makes it clear what's happening:

```shell
echo "file_name=$(ruby -r /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/.github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.write_output_file)" >> $GITHUB_ENV
```